diff -uNr glib-2.49.3/configure.ac glib-2.49.3-patched/configure.ac
--- glib-2.49.3/configure.ac	2016-07-17 19:18:08.000000000 +0200
+++ glib-2.49.3-patched/configure.ac	2016-07-17 22:39:15.535344842 +0200
@@ -385,14 +385,14 @@
 # option to specify python interpreter to use; this just sets $PYTHON, so that
 # we will fallback to reading $PYTHON if --with-python is not given, and
 # python.m4 will get the expected input
-AC_ARG_WITH(python,
-            AS_HELP_STRING([--with-python=PATH],
-                           [Path to Python interpreter; searches $PATH if only a program name is given; if not given, searches for a few standard names such as "python3" or "python2"]),
-            [PYTHON="$withval"], [])
-if test x"$PYTHON" = xyes; then
-  AC_MSG_ERROR([--with-python option requires a path or program argument])
-fi
-AM_PATH_PYTHON(2.5,,PYTHON="/usr/bin/env python2.5")
+# AC_ARG_WITH(python,
+#             AS_HELP_STRING([--with-python=PATH],
+#                            [Path to Python interpreter; searches $PATH if only a program name is given; if not given, searches for a few standard names such as "python3" or "python2"]),
+#             [PYTHON="$withval"], [])
+# if test x"$PYTHON" = xyes; then
+#   AC_MSG_ERROR([--with-python option requires a path or program argument])
+# fi
+# AM_PATH_PYTHON(2.5,,PYTHON="/usr/bin/env python2.5")
 
 
 dnl ***********************
@@ -3434,7 +3434,6 @@
 gobject/tests/Makefile
 gthread/Makefile
 gio/Makefile
-gio/gdbus-2.0/codegen/Makefile
 gio/gdbus-2.0/codegen/config.py
 gio/gnetworking.h
 gio/xdgmime/Makefile
diff -uNr glib-2.49.3/docs/reference/gio/Makefile.am glib-2.49.3-patched/docs/reference/gio/Makefile.am
--- glib-2.49.3/docs/reference/gio/Makefile.am	2016-07-01 22:07:05.000000000 +0200
+++ glib-2.49.3-patched/docs/reference/gio/Makefile.am	2016-07-17 22:41:07.234554948 +0200
@@ -176,7 +176,6 @@
 	gsettings.1			\
 	gresource.1			\
 	gdbus.1				\
-	gdbus-codegen.1			\
 	gio.1				\
 	$(NULL)
 
diff -uNr glib-2.49.3/gio/Makefile.am glib-2.49.3-patched/gio/Makefile.am
--- glib-2.49.3/gio/Makefile.am	2016-07-17 05:29:01.000000000 +0200
+++ glib-2.49.3-patched/gio/Makefile.am	2016-07-17 22:39:15.536344835 +0200
@@ -1,6 +1,6 @@
 include $(top_srcdir)/glib.mk
 
-SUBDIRS = gdbus-2.0/codegen
+SUBDIRS =
 
 if OS_UNIX
 SUBDIRS += xdgmime
diff -uNr glib-2.49.3/gio/tests/gdbus-object-manager-example/Makefile.am glib-2.49.3-patched/gio/tests/gdbus-object-manager-example/Makefile.am
--- glib-2.49.3/gio/tests/gdbus-object-manager-example/Makefile.am	2016-07-01 21:54:29.000000000 +0200
+++ glib-2.49.3-patched/gio/tests/gdbus-object-manager-example/Makefile.am	2016-07-17 22:39:15.536344835 +0200
@@ -11,10 +11,8 @@
 	gdbus-example-objectmanager-generated-org.gtk.GDBus.Example.ObjectManager.Cat.xml	\
 	$(NULL)
 
-$(GDBUS_GENERATED) : gdbus-example-objectmanager.xml Makefile $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen
-	$(AM_V_GEN) UNINSTALLED_GLIB_SRCDIR=$(top_srcdir) \
-		UNINSTALLED_GLIB_BUILDDIR=$(top_builddir) \
-		$(PYTHON) $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen \
+$(GDBUS_GENERATED) : gdbus-example-objectmanager.xml Makefile
+	$(AM_V_GEN) gdbus-codegen \
 		--interface-prefix org.gtk.GDBus.Example.ObjectManager. \
 		--c-namespace Example \
 		--c-generate-object-manager \
diff -uNr glib-2.49.3/gio/tests/Makefile.am glib-2.49.3-patched/gio/tests/Makefile.am
--- glib-2.49.3/gio/tests/Makefile.am	2016-07-17 03:20:53.000000000 +0200
+++ glib-2.49.3-patched/gio/tests/Makefile.am	2016-07-17 22:39:15.536344835 +0200
@@ -449,10 +449,8 @@
 
 BUILT_SOURCES += gdbus-test-codegen-generated.c gdbus-test-codegen-generated.h
 gdbus-test-codegen.o: gdbus-test-codegen-generated.h
-gdbus-test-codegen-generated.h: test-codegen.xml Makefile $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen
-	$(AM_V_GEN) UNINSTALLED_GLIB_SRCDIR=$(top_srcdir) \
-		UNINSTALLED_GLIB_BUILDDIR=$(top_builddir) \
-		$(PYTHON) $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen \
+gdbus-test-codegen-generated.h: test-codegen.xml Makefile
+	$(AM_V_GEN) gdbus-codegen \
 		--interface-prefix org.project. \
 		--generate-c-code gdbus-test-codegen-generated \
 		--c-generate-object-manager \
